components/BottomNavigation/tests/unit/BottomNavigationItemViewTests.m

```diff
@@ -127,4 +131,42 @@ - (void)testSetSelectedItemTintColorUpdatesInkColor {
   XCTAssertNotEqualObjects(item2.inkView.inkColor, item2DefaultInkColor);
 }
 
+- (void)testBadgeAndIconHaveSameOriginY {
```

Comment:
Please add a test for the new method.  Once that's done, the test for the actual badge position can be simply to make sure it's positioned correctly for RTL rather than the exact offset from the frame.